sesame2spiner/generate_files.cpp

```diff
@@ -273,34 +274,110 @@ void getMatBounds(int i, int matid, const SesameMetadata &metadata, const Params
               << "shrinkleBounds > 0 and siemin or siemax set" << std::endl;
   }
 
-  int ppdRho = params.Get("numrho/decade", PPD_DEFAULT);
-  int numRhoDefault = getNumPointsFromPPD(rhoMin, rhoMax, ppdRho);
+  int ppdRho = params.Get("numrho/decade", PPD_DEFAULT_RHO);
+  int numRhoDefault = Bounds::getNumPointsFromPPD(rhoMin, rhoMax, ppdRho);
 
-  int ppdT = params.Get("numT/decade", PPD_DEFAULT);
-  int numTDefault = getNumPointsFromPPD(TMin, TMax, ppdT);
+  int ppdT = params.Get("numT/decade", PPD_DEFAULT_T);
+  int numTDefault = Bounds::getNumPointsFromPPD(TMin, TMax, ppdT);
 
-  int ppdSie = params.Get("numSie/decade", PPD_DEFAULT);
-  int numSieDefault = getNumPointsFromPPD(sieMin, sieMax, ppdSie);
+  int ppdSie = params.Get("numSie/decade", PPD_DEFAULT_T);
+  int numSieDefault = Bounds::getNumPointsFromPPD(sieMin, sieMax, ppdSie);
 
   int numRho = params.Get("numrho", numRhoDefault);
   int numT = params.Get("numT", numTDefault);
   int numSie = params.Get("numsie", numSieDefault);
 
-  Real rhoAnchor = metadata.normalDensity;
-  Real TAnchor = 298.15;
+  const Real TAnchor = 298.15;
+  Real rhoAnchor = params.Get("rho_fine_center", metadata.normalDensity);
+  if (std::isnan(rhoAnchor) || rhoAnchor <= 0 || rhoAnchor > 1e8) {
```

Comment:
Do we encounter `NaN`s in the `normalDensity` parameter?

Reply:
Probably not but I am concerned about what happens if it's not defined for a given material. For example, a custom material someone tries to use. That's just me being cautious.